helpdesk/email.py

```diff
@@ -225,6 +226,101 @@ def imap_sync(q, logger, server):
     server.logout()
 
 
+def imap_oauth_sync(q, logger, server):
+    """
+    IMAP eMail server with OAUTH authentication.
+    Only tested against O365 implementation
+
+    Uses HELPDESK OAUTH Dict in Settings.
+
+    """
+
+    try:
+        logger.debug("Start Mailbox polling via IMAP OAUTH")
+
+        client = oauth2lib.BackendApplicationClient(
+            client_id=settings.HELPDESK_OAUTH["client_id"],
+            scope=settings.HELPDESK_OAUTH["scope"],
+        )
+
+        oauth = requests_oauthlib.OAuth2Session(client=client)
+        token = oauth.fetch_token(
+            token_url=settings.HELPDESK_OAUTH["token_url"],
+            client_id=settings.HELPDESK_OAUTH["client_id"],
+            client_secret=settings.HELPDESK_OAUTH["secret"],
+            include_client_id=True,
+        )
+
+        # TODO: Somehow link this to the debug level set within Django settings logging
+        server.debug = 4
```

Comment:
The IMAP library has a multi-level logging setting that does not really correlate to the Django LOG_LEVEL values.
Add a settings entry to allow the user set the debug level for IMAP independently of the Django log level:
`HELPDESK_IMAP_DEBUG_LEVEL = int(os.environ.get("HELPDESK_IMAP_DEBUG_LEVEL", "0"))`

Then change this statement to:
`server.debug = settings.HELPDESK_IMAP_DEBUG_LEVEL`